package-res/resources/web/pentaho/util/arg.js

```diff
@@ -20,23 +20,73 @@ define([
 
   var A_slice = Array.prototype.slice;
 
+  /**
+   * @name pentaho.util.arg
+   * @class
+   * @amd pentaho/util/arg
+   *
+   * @classDesc Util methods for keyword args.
+   *
+   * The AMD module returns the arg utils.
+   *
+   * @description Performs actions on keyword args.
+   */
   return {
+    /**
+     * Provides the optional property value of a given object or the given default value if the given object does
+     * not contain the sought after property value.
+     *
+     * @param {?object} [obj] The object to search for given propName param.
+     * @param {String} [propName] Property name to find in obj param.
+     * @param {any} [defaultValue] The default value to return if propName param is not found. The default value of this param is undefined and is considered specified when its value is not a nully value. {@link Nully}
+     *
+     * @return {any} The found optional property value or the default value.
+     */
     optional: function(o, p, dv) {
       var v;
       return o && (v = o[p]) != null ? v : dv;
     },
 
+    /**
+     * Provides the defined property value of a given object or the given default value if the given object does
+     * not contain the sought after property value.
+     *
+     * @param {?object} [obj] The object to search for given propName param.
+     * @param {String} [propName] Property name to find in obj param.
+     * @param {any} [defaultValue] The default value to return if propName param is not found. The default value of this param is undefined and is considered specified when its value is not a nully value. {@link Nully}
+     *
+     * @return {any} The found defined property value (a property is considered defined if its value is not undefined) or the default value.
+     */
     defined: function(o, p, dv) {
       var v;
       return o && (v = o[p]) !== undefined ? v : dv;
     },
 
+    /**
+     * Provides the required property value of a given object.
+     *
+     * @param {?object} [obj] The object to search for given propName param.
+     * @param {String} [propName] Property name to find in obj param.
+     * @param {?string} [pscope] The name of the argument where the obj param is received in the caller.
+     *
+     * @return {any} The found required property value.
+     * @throws {Error} Argument required. The o param must contain the p param property.
+     */
     required: function(o, p, pscope) {
       var v;
       if(o && (v = o[p]) != null) return v;
       throw error.argRequired((pscope ? (pscope + ".") : "") + p);
     },
 
+    /**
+     * Slices the provided array.
+     *
+     * @param {Object} [args] Array of anything.
+     * @param {Number} [start] The index of the args array to start the slice.
+     * @param {Number} [end] The index of the ars param array to end the slice at.
+     *
```

Comment:
args is not an optional parameter.
start is optional with a default value of 0
end is optional, defaults to...